### PR TITLE
[14.0][ADD] stock_cycle_count_abc

### PR DESCRIPTION
**Pull request by Freni-OSI**
This module allows you to implement ABC Analysis and configure cycle count rules using the ABC group.
Issue #1364

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Freni-OSI Thanks for this. Shouldn't it be named 'stock_cycle_count_abc' to follow the same naming ?

**Comment by smangukiya** [comment]
@Freni-OSI Can you please improve the code coverage %? Thanks

File: stock_inventory_abc/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright (C) 2022 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Inventory ABC",
+    "summary": "This module allows you to implement ABC Analysis \
+and configure cycle count rules using the ABC group.",
+    "version": "14.0.1.0.0",
+    "maintainers": ["opensourceintegrators"],
```
Comment (rousseldenis) [inline stock_inventory_abc/__manifest__.py]: This should be a github login

File: stock_inventory_abc/models/stock_cycle_count.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                "exclude_sublocation": True,
+                "product_ids": [(6, 0, all_product_list)],
+            }
+            return abc_data
+        return super()._prepare_inventory_adjustment()
+
+    def action_create_inventory_adjustment(self):
+        res = super().action_create_inventory_adjustment()
+        if self.cycle_count_rule_id.rule_type == "abc":
+            draft_inventory = self.env["stock.inventory"].search(
```
Comment (rousseldenis) [inline stock_inventory_abc/models/stock_cycle_count.py]: Is this really wanted to change all draft inventories state ?

File: stock_inventory_abc/models/stock_cycle_count_rule.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        if self.rule_type == "abc":
+            proposed_cycle_counts = self._compute_rule_abc(locs)
+            return proposed_cycle_counts
+        return super().compute_rule(locs)
+
+    @api.model
+    def _compute_rule_abc(self, locs):
+        cycle_counts = []
+        for loc in locs:
+            latest_inventory_date = (
```
Comment (rousseldenis) [inline stock_inventory_abc/models/stock_cycle_count_rule.py]: Searches in loops should be avoided.

**Comment by Freni-OSI** [comment]
> stock_cycle_count_abc

@max3903 what are your thoughts on this?

**Comment by Freni-OSI** [comment]
Can you please review this PR?
CC: @max3903 @smangukiya @rousseldenis

**Comment by rousseldenis** [comment]
@Freni-OSI @max3903 IMHO this is duplicate functionality of product_abc_classification.

https://github.com/OCA/product-attribute/blob/13.0/product_abc_classification/models/abc_classification_level.py

You need to reuse the link on the product to the abc classification instead of reitroducing a new model (abc_id field).

@MiquelRForgeFlow @CarlosRoca13

**Comment by rousseldenis** [comment]
@Freni-OSI @max3903 @dreispt What's the status of this ?

**Pull request closed without merging**